File: Easy/Alternate positive and negative numbers/alternate-positive-and-negative-numbers.py
#User function Template for python3

class Solution:
    def rearrange(self, arr, n):
        p, n, ans = [], [], []
        for i in arr:
            if i >= 0:
                p.append(i)
            else:
                n.append(i)
        
        i, j = 0, 0
        
        
        while i < len(p) and j < len(n):
            ans.append(p[i])
            i += 1
            ans.append(n[j])
            j += 1
        
        
        ans.extend(p[i:])
        ans.extend(n[j:])
        
        
        arr[:] = ans[:]



#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
	tc = int(input())
	while tc > 0:
		n = int(input())
		arr = list(map(int, input().strip().split()))
		ob = Solution()
		ob.rearrange(arr, n)
		for x in arr:
			print(x, end=" ")
		print()
		tc -= 1

# } Driver Code Ends

File: Difficulty: Easy/Alternate positive and negative numbers/alternate-positive-and-negative-numbers.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int tc = Integer.parseInt(br.readLine().trim());
        while (tc-- > 0) {
            String input = br.readLine();
            String[] inputArray = input.split("\\s+");
            ArrayList<Integer> arr = new ArrayList<>();

            for (String s : inputArray) {
                arr.add(Integer.parseInt(s));
            }

            new Solution().rearrange(arr);
            for (int num : arr) {
                System.out.print(num + " ");
            }
            System.out.println();
        }
    }
}

// } Driver Code Ends


// User function Template for Java


class Solution {
    void rearrange(ArrayList<Integer> arr) {
        // code here
        Queue<Integer> pNum = new LinkedList<Integer>();
        Queue<Integer> nNum = new LinkedList<Integer>();
        for(Integer i : arr)
        {
            if(i >= 0)
            {
                pNum.add(i);
            }
            else
            {
                nNum.add(i);
            }
        }
        arr.clear(); // clear all the elements of arr
        while(!pNum.isEmpty() || !nNum.isEmpty())
        {
            if(!pNum.isEmpty())
            {
                Integer ele = pNum.poll(); 
                arr.add(ele);
            }
            if(!nNum.isEmpty())
            {
                Integer ele = nNum.poll();
                arr.add(ele);
            }
        }
    }
}